File: backend/schemas/claims.py
"""Pydantic v2 schemas for claims endpoints."""

from typing import Optional, List
from pydantic import BaseModel


class ClaimLineSchema(BaseModel):
    """Single claim line item."""
    claim_line_id: int
    claim_id: int
    line_number: int
    cpt_code: str
    procedure_description: Optional[str] = None
    units: str  # Decimal as string
    unit_price: str  # Decimal as string
    line_billed_amount: str  # Decimal as string
    line_status: str


class StatusHistoryEntry(BaseModel):
    """Claim status transition history record."""
    history_id: int
    claim_id: int
    previous_status_code: Optional[str] = None
    new_status_code: str
    transition_timestamp: str
    transition_reason: Optional[str] = None
    actor_reference: str


class ClaimSummary(BaseModel):
    """Claim list item (paginated results)."""
    claim_id: int
    claim_reference: str
    encounter_id: int
    patient_id: int
    billing_provider_id: int
    payer_id: int
    current_status_code: str
    total_billed_amount: str  # Decimal as string
    submission_date: str
    adjudication_date: Optional[str] = None
    is_reconciled: bool


class ClaimDetail(BaseModel):
    """Full claim detail with financial summary."""
    claim_id: int
    claim_reference: str
    encounter_id: int
    patient_id: int
    billing_provider_id: int
    payer_id: int
    current_status_code: str
    total_billed_amount: str
    submission_date: str
    adjudication_date: Optional[str] = None
    is_reconciled: bool
    lines: Optional[List[ClaimLineSchema]] = None
    total_paid: Optional[str] = None
    total_adjusted: Optional[str] = None
    total_denied: Optional[str] = None

File: frontend/src/types/claims.ts
export interface ClaimSummary {
  claim_id: number;
  claim_reference: string;
  encounter_id: number;
  patient_id: number;
  billing_provider_id: number;
  payer_id: number;
  current_status_code: string;
  total_billed_amount: string;
  submission_date: string;
  adjudication_date: string | null;
  is_reconciled: boolean;
}

export interface ClaimLine {
  claim_line_id: number;
  claim_id: number;
  line_number: number;
  cpt_code: string;
  procedure_description?: string | null;
  units: string;
  unit_price: string;
  line_billed_amount: string;
  line_status: string;
}

export interface StatusHistoryEntry {
  history_id: number;
  claim_id: number;
  previous_status_code?: string | null;
  new_status_code: string;
  transition_timestamp: string;
  transition_reason?: string | null;
  actor_reference: string;
}

export interface ClaimDetail extends ClaimSummary {
  lines?: ClaimLine[];
  total_paid?: string;
  total_adjusted?: string;
  total_denied?: string;
}

export interface ClaimFilters {
  status?: string;
  claim_reference?: string;
  payer_id?: number;
  provider_id?: number;
  patient_id?: number;
  start_date?: string;
  end_date?: string;
  is_reconciled?: boolean;
  page?: number;
  page_size?: number;
}
